Src/WitsmlExplorer.Api/Workers/DownloadLogDataWorker.cs

```diff
@@ -114,4 +149,361 @@ private string GetReportBody(ICollection<Dictionary<string, LogDataValue>> repor
+        WriteWellParameter(writer, "SAPD", "", "", "Elevation of Depth Reference (LMF) above Permanent Datum", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SEPD", "", "", "Elevation of Permanent Datum (PDAT) above Mean Sea Level", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SEKD", "", "", "Elevation of Kelly Bushing above Permanent Datum", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SEDF", "", "", "Elevation of Drill Floor above Permanent Datum", maxColumnLength, maxDataLenght);
+    }
+
+    private void WriteCommonParameter(StringWriter writer, string nameOfParameter, string data, string description, int maxColumnLength, int maxDataLenght)
+    {
+        var line = new StringBuilder();
+        line.Append(nameOfParameter);
+        if (maxColumnLength - nameOfParameter.Length > 0)
+        {
+            line.Append(new string(' ', maxColumnLength - nameOfParameter.Length));
+        }
+        line.Append($" {data}");
+        if (maxColumnLength - data.Length > 0)
+        {
+            line.Append(new string(' ', maxColumnLength - data.Length));
+        }
+        line.Append(new string(' ', maxDataLenght));
+        line.Append($":{description}");
+        writer.WriteLine(line.ToString());
+    }
+    private void WriteWellParameter(StringWriter writer, string nameOfParemeter, string unit,
+        string data, string description, int maxColumnLength, int maxDataLength)
+    {
+        var line = new StringBuilder();
+        line.Append(nameOfParemeter);
+        if (maxColumnLength - nameOfParemeter.Length > 0)
+        {
+            line.Append(new string(' ', maxColumnLength - nameOfParemeter.Length));
+        }
+        line.Append($".{unit}");
+        if (maxColumnLength - unit.Length - 1 > 0)
+        {
+            line.Append(new string(' ', maxColumnLength - unit.Length - 1));
+        }
+        line.Append(data);
+        if (data == null)
+        {
+            line.Append(new string(' ', maxDataLength));
+        }
+        else if (maxDataLength - data.Length > 0)
+        {
+            line.Append(new string(' ', maxDataLength - data.Length));
+        }
+        line.Append($" :{description}");
+        writer.WriteLine(line.ToString());
+    }
+
+
+    private void WriteDataSection(StringWriter writer,
+        ICollection<Dictionary<string, LogDataValue>> data, ICollection<CurveSpecification> curveSpecifications, Dictionary<string, int> columnsLength)
+    {
+        foreach (var row in data)
+        {
+            var line = new StringBuilder();
+            int i = 0;
+            foreach (var curveSpecification in curveSpecifications)
+            {
+                var cell = row.TryGetValue(curveSpecification.Mnemonic, out LogDataValue value)
+                        ? value.Value.ToString()
+                        : string.Empty;
```

Comment:
I think we need to use the null value instead of an empty string for LAS export. Otherwise, we wouldn't be able to import it using our application afterwards. You can find it in the corresponding logObject.LogCurveInfo.NullValue, but it is often not specified. If it is not, then we should use the default one (as from CommonConstants).

![image](https://github.com/user-attachments/assets/92fb0a65-1e35-4464-959b-ad74ea796329)

As seen above, we need a value in each cell.

Reply:
done